sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs

```diff
@@ -16,10 +17,32 @@ public class MonitorQueryTestEnvironment : TestEnvironment
         public string MetricsNamespace => GetRecordedVariable("METRICS_RESOURCE_NAMESPACE");
         public string MonitorIngestionEndpoint => GetOptionalVariable("METRICS_INGEST_SUFFIX") ?? "ods.opinsights.azure.com";
         public string MetricsIngestionEndpoint => GetOptionalVariable("METRICS_INGEST_SUFFIX") ?? "monitoring.azure.com";
-        public Uri LogsEndpoint => new(GetRecordedVariable("LOGS_ENDPOINT"));
         public Uri MetricsEndpoint => new(ResourceManagerUrl);
         public string ResourceId => GetRecordedVariable("RESOURCE_ID");
         public string WorkspacePrimaryResourceId => GetRecordedVariable("WORKSPACE_PRIMARY_RESOURCE_ID");
         public string WorkspaceSecondaryResourceId => GetRecordedVariable("WORKSPACE_SECONDARY_RESOURCE_ID");
+        public Uri LogsEndpoint => new(GetEndpoint());
+
+        private Dictionary<string, string> regions = new Dictionary<string, string>()
+        {
+            { "AzureCloud", "https://api.loganalytics.io/v1" },
+            { "AzureChinaCloud", "https://api.loganalytics.azure.cn/v1" },
+            { "AzureGovernmentCloud", "https://api.loganalytics.us/v1" }
+        };
+
+        private string ENV_MONITOR_ENVIRONMENT = "MONITOR_ENVIRONMENT";
+        private string GetEndpoint()
+        {
+            string endpoint = "";
+            if (regions.TryGetValue(Environment.GetEnvironmentVariable(ENV_MONITOR_ENVIRONMENT), out string region))
```

Comment:
How does this environment variable get set?

Reply:
Ah it is here - https://github.com/Azure/azure-sdk-for-net/blob/main/eng/common/TestResources/New-TestResources.ps1#L273

Reply:
thank you!